public/modules/nav/navController.js

```diff
@@ -2,8 +2,25 @@
  * Created by brillwill on 16/9/14.
  */
 app.controller("navController", ["$scope", "$rootScope", function($scope, $rootScope){
+	$scope.hasLoggedIn = false;
+
     $scope.goLogin = function () {
         $rootScope.$state.go("nav.login");
-    }
+    };
+
+    $rootScope.$on('loggedIn',function(){
+    	$scope.hasLoggedIn = true;
+    });
+
+    $rootScope.$on('loggedOut',function(){
+    	$rootScope.currentUser = null;
```

Comment:
This logic should not be here. navController is only responsible for model and view update, but not modify model here, as you have loginservice. You can have it in loginService to do the real login/logout.